.github/actions/with-docker/Dockerfile

```diff
@@ -17,4 +20,8 @@ RUN groupadd -g $GROUP_ID $GROUP && useradd -m -u $USER_ID -s /bin/sh -g $GROUP
 USER $USER:$GROUP
 
 ENV PATH=/home/$USER/.local/bin:$PATH
-RUN curl -sSL https://install.python-poetry.org | python3 -
+
+ARG UV2NIX_REVISION
+RUN UV_VERSION=$(curl -s https://raw.githubusercontent.com/pyproject-nix/uv2nix/${UV2NIX_REVISION}/pkgs/uv-bin/srcs.json | jq -r .version) \
+    && [[ "$UV_VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] \
+    && curl -LsSf https://astral.sh/uv/$UV_VERSION/install.sh | sh && uv --version
```

Comment:
Here, I think `UV_VERSION` should be passed directly, with the (currently) most recent `uv` version as default value. The `UV_VERSION` then can be resolved externally.